Source/Core/AudioCommon/WASAPIStream.h

```diff
@@ -0,0 +1,50 @@
+// Copyright 2018 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#ifdef _WIN32
+// clang-format off
+#include <Windows.h>
+#include <mmreg.h>
+// clang-format on
+#endif
+
+#include <atomic>
+#include <string>
+#include <thread>
+#include <vector>
+
+#include "AudioCommon/SoundStream.h"
+
+struct IAudioClient;
+struct IAudioRenderClient;
+struct IMMDevice;
+struct IMMDeviceEnumerator;
+
+class WASAPIStream final : public SoundStream
+{
+#ifdef _WIN32
+public:
+  explicit WASAPIStream();
+  ~WASAPIStream();
+  static bool isValid();
+  bool Init() override;
+  bool SetRunning(bool running) override;
+  void SoundLoop() override;
+  std::vector<std::string> GetAvailableDevices();
+  IMMDevice* GetDeviceByName(std::string name);
+
+private:
+  u32 m_frames_in_buffer;
```

Comment:
These class variables (this and the atomic bools) should likely have initial values.